Comorphisms/HPAR2CASL.hs

```diff
@@ -0,0 +1,454 @@
+{-# LANGUAGE MultiParamTypeClasses, TypeSynonymInstances, FlexibleInstances #-}
+{- |
+Module      :  ./Comorphisms/HPAR2CASL.hs
+Description :  encoding of HPAR to FOL
+Copyright   :  (c) M. Codescu, IMAR, 2018
+License     :  GPLv2 or higher, see LICENSE.txt
+
+Maintainer  :  mscodescu@gmail.com
+Stability   :  provisional
+Portability :  non-portable (imports Logic.Logic)
+
+The encoding comorphism from HPAR to FOL.
+
+-}
+
+module Comorphisms.HPAR2CASL where
+
+import Logic.Logic
+import Logic.Comorphism
+import qualified Data.Set as Set
+import qualified Data.Map as Map
+import Common.ProofTree
+import Common.Result
+import Common.AS_Annotation
+import Common.Id
+import qualified Common.Lib.MapSet as MapSet
+
+
+import Control.Monad (foldM)
+import Data.List(partition)
+
+-- HPAR
+import qualified HPAR.Logic_HPAR as HLogic
+import qualified HPAR.AS_Basic_HPAR as HBasic
+import qualified HPAR.Sign as HSign
+import qualified RigidCASL.Sign as RSign
+import qualified HPAR.Morphism as HMorphism
+import qualified HPAR.StaticAna as HAna
+
+-- CASL
+import qualified CASL.Logic_CASL as CLogic
+import qualified CASL.AS_Basic_CASL as CBasic
+import qualified CASL.Sign as CSign
+import qualified CASL.Morphism as CMor
+import CASL.Sublogic 
+import qualified CASL.Induction as CInd
+
+-- base comorphism
+import qualified Comorphisms.CASL2SubCFOL as BaseCom
+
+-- | The identity of the comorphism
+data HPAR2CASL = HPAR2CASL deriving (Show)
+
+instance Language HPAR2CASL
+
+instance Comorphism HPAR2CASL
+               HLogic.HPAR ()
+               HBasic.H_BASIC_SPEC HBasic.HFORMULA HBasic.H_SYMB_ITEMS 
+               CBasic.SYMB_MAP_ITEMS
+               HSign.HSign HMorphism.HMorphism CSign.Symbol CMor.RawSymbol () 
+               CLogic.CASL CASL_Sublogics
+               CLogic.CASLBasicSpec CBasic.CASLFORMULA CBasic.SYMB_ITEMS CBasic.SYMB_MAP_ITEMS
+               CSign.CASLSign
+               CMor.CASLMor
+               CSign.Symbol CMor.RawSymbol ProofTree where
+    sourceLogic HPAR2CASL = HLogic.HPAR
+    sourceSublogic HPAR2CASL = ()
+    targetLogic HPAR2CASL = CLogic.CASL
+    mapSublogic HPAR2CASL _ = Just cFol
+    map_theory HPAR2CASL = mapTheory
+    map_morphism HPAR2CASL = undefined
+    map_sentence HPAR2CASL =  undefined
+    map_symbol HPAR2CASL _ = undefined
+    -- has_model_expansion HPAR2CASL = True 
+    -- @Till: this makes proofs by translation available, but on a weird path
```

Comment:
yes, the path is weird. Please comment this in and add your comorphism to `Logic.KnownProvers.spassComorphisms`.

Reply:
Thanks, it works now.